src/runtime/entries/cloudflare.ts

```diff
@@ -71,9 +73,20 @@ const baseURLModifier = (request: Request) => {
 
 function normalizeOutgoingHeaders(
   headers: Record<string, string | string[] | undefined>
-) {
-  return Object.entries(headers).map(([k, v]) => [
-    k,
-    Array.isArray(v) ? v.join(",") : v,
-  ]);
+): Headers {
+  const result = new Headers();
+
+  for (const [k, v] of Object.entries(headers)) {
+    if (k === "set-cookie") {
+      for (const cookie of splitCookiesString(v)) {
```

Comment:
We have recently introduced [same util](https://github.com/unjs/h3/blob/80f2c611918a17d1c303636924ce39ea7f246489/src/utils/internal/cookies.ts#L10) in h3. It might be one less dependency and bundle size if we expose it from h3 package.

Reply:
Thanks, what is proper way to sync versions of h3 and current nitro? Should we wait for h3 version tag or?

Reply:
Sure, releasing next version until this weekend. (at the same time thinking about your other comments, we need to really fix header normalization across entries)

Reply:
As i see normalize function looks same for all other entries, but i think something is different on provider level, i suppose that some of same do cookie splitting themselves. We can drop normalize function to utils and use it every entry until make changes with type change for `HadersObject`.

Reply:
Indeed there are slight behavior diferences. Also we never properly tested header behavior on [nitro-deploys](https://github.com/unjs/nitro-deploys) 😓

Reply:
Hello, @pi0. Do you have any plans for releasing new versions soon? Gonna continue work on module and project, but this issue blocks me testing any env except local one :)